### PR TITLE
feat(DTFS2-7450): add e2e tests for cancel cancellation

### PR DESCRIPTION
**Pull request by MarRobSoftwire**
## Introduction :pencil2:
The cancel cancellation page needs e2e test coverage

## Resolution :heavy_check_mark:
Add e2e tests

## Miscellaneous :heavy_plus_sign:
Update non-PIM user e2e tests as they weren't testing properly

## Note ⚠️ 
Add some tests that rely on behaviour implmented by #3649

### CODE REVIEW

File: e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      });
+    });
+
+    describe('when a cancellation is in draft', () => {
+      beforeEach(() => {
+        cy.visit(relative(`/case/${dealId}/deal`));
+
+        caseDealPage.cancelDealButton().click();
+
+        cy.keyboardInput(reasonForCancellingPage.reasonForCancellingTextBox().clear(), 'xxx');
```
Comment (ttbarnes) [inline e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js:59]: No need for `.clear()` here as keyboardInput does this for us

File: e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            cy.clickCancelLink();
+          });
+
+          it(`back link navigates to ${page} page`, () => {
+            cy.clickBackLink();
+
+            cy.url().should('eq', relative(`/case/${dealId}/cancellation/${page}`));
+          });
+
+          it(`no, go back button navigates to ${page} page`, () => {
```
Comment (ttbarnes) [inline e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js:115]:
```suggestion
          it(`"no, go back" button navigates to ${page} page`, () => {
```

File: e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+          cancelCancellationPage.yesCancelButton().click();
+
+          cy.url().should('eq', relative(`/case/${dealId}/deal`));
+        });
+
+        it('yes button wipes the cancellation data', () => {
+          cancelCancellationPage.yesCancelButton().click();
+
+          caseDealPage.cancelDealButton().click();
+          reasonForCancellingPage.reasonForCancellingTextBox().should('have.value', '');
```
Comment (ttbarnes) [inline e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js:98]: Do we need to test for any other fields related to cancellation - date fields etc?

File: e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        });
```
Comment (ttbarnes) [inline e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js:99]: Does the deal status change after cancellation? If so, would be great to check the status hasn't changed after cancelling the cancellation

**Review by ttbarnes — CHANGES_REQUESTED** [review]

File: e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js (hunk 3 above)
Comment (MarRobSoftwire) [reply e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js:98]: Adding a `'when all the values are provided'` describe block to check the request and effective from dates

File: e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js (hunk 4 above)
Comment (MarRobSoftwire) [reply e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/cancel-cancellation.spec.js:99]: Not yet - this is a future PR

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**